### DISCUSSION TITLE
Windows Terminal v1.18.2681.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#16036] [Announcements]
Windows Terminal 1.18 is here, and with it comes the coolest set of features ever:

* **You can finally pull tabs out of Terminal windows!** You can also put other tabs back into other Terminal
  windows. It's a whole thing.
* A new _new_ rendering engine that supports arbitrary glyph overhangs (including some new fixes backported from
  1.19!)
* An experimental (opt-in) right-click context menu (profile setting `{profile}.rightClickContextMenu`)

Please see the following release notes for additional details.

- [Windows Terminal Preview v1.18.1421.0](https://github.com/microsoft/terminal/releases/tag/v1.18.1421.0)
- [Windows Terminal Preview v1.18.1462.0](https://github.com/microsoft/terminal/releases/tag/v1.18.1462.0)

Note that scrollbar marks and shell integration features are still considered experimental and
_are not included in this stable release!_

> **Note**
> This version began rolling out to the Dev and Canary Windows Insider channels on 2023-09-26. Other channels will be released
> as our reliability numbers indicate that we haven't broken anything. If you want to update to this release directly, download it below!

### Why are there so many packages? How do I choose?

Please visit our page documenting the different [Windows Terminal Distributions]!

This release also includes the following changes and bug fixes backported from Preview 1.19:

## Changes

* `wt -- foo` will once again inherit the environment from its parent (#15897)
   * You can customize this behavior with `wt --inheritEnvironment` or `wt --reloadEnvironment` on a per-tab/per-pane basis.
   * You can also override this behavior with the profile setting `{profile}.compatibility.reloadEnvironmentVariables` (boolean, default `true`)
* We no longer load the Settings editor library on launch, saving you tens of kilobytes of memory (#15628) (#15631)
* You can now run profiles as Admin from the new tab menu by right-clicking them (#15679) (thanks @jamespack!)
* The profile preview in settings now shows you what a "powerline" font will look like when one is selected (#15365)
* We actually did fix the homoglyph thing this time; hovering a URL that is pretending to be someone else through funny characters will see right through it (#15488)
* For those of you debugging Terminal/the console with ETW, you can now group all API calls by originating process (#15737)
* If you are using Preview, Canary or Dev we will now set an appropriate JSON `$schema` so you can auto-complete new settings (#15856)

### Accessibility

* We've added automation property names to the 'Delete Color Scheme' button (#15994)
* The key chord indicator in the command palette now has a background color (#15677) (thanks @RickleAndMortimer!)
* Settings now groups elements in their containers for screen readers (#15756)
* Screen reader users can now determine which color scheme is the current default (#15486)
* Default Terminal and Color Scheme ComboBoxes no longer crop at 200% text scale (#15762)
* Screen readers will now announce successful pane and tab moves (#15771)

### VT and terminal emulation

* Hard Reset (`RIS`) no longer disables Win32 input mode or focus event mode for pty clients (like Terminal) (#15476) (thanks @j4james!)

### Usability and UI

* We will no longer reload the settings as often when various OS knobs and switches change (#16004)
* You can once again restart the first pane in a split with <kbd>Enter</kbd> after it exits (#16001)
* All actions now remember which pane originated them, so things like "export text" and "Move tab to new window" and "Duplicate" no longer summon one of the Old Ones (#15773)
* The about dialog no longer tells you that you've got an update available to the same version you already have (#15378)
* When the tab close button is disabled, it will no longer randomly reappear to taunt you (#15914)
* The launch parameters (position, size) now properly support negative numbers (#15941) (thanks @codecruisedor!)

### New Rendering Engine

* We no longer leave cursors all over the place (#15904)
* Fonts like JetBrains Mono that render their ligatures in reverse now appear in the right colors (#15810)
* We no longer invalidate more of the screen than we need to (#15929)
* It now works properly on Windows versions earlier than 10.0.14393.0 (#15485)
* Font features and axes work once again! (#15912)
* We no longer crash when trying to render to a 0x0 surface (#15615)
* DRCS soft fonts that are invalid (or have disappeared, or were never enabled) no longer crash the Terminal (#15889)
* Both rendering engines have been prepared for horizontal scrolling (???) (#15707)

### Reliability

* We will no longer crash when checking for updates without a network (#16002)
* If you're the one person who marks `HKCU\Console` as read-only, we won't crash during save for you (#15916)
* Actions with an empty `keys` array no longer cause unexpected kinetic disassembly (#16003)
* Duplicating `elevate: true` tabs has been fixed to not blast terminal off into space (#15548)
* The backing console no longer hangs when resizing while scrolling (#15618)
* Instead of leaking memory for each window on Windows 10, we'll cache old windows and recycle them as needed (#15424)
   * We introduced the leak to fix a crash in closing windows on Windows 10; this allows us to still leak the window but still be able to reclaim them.

### Schema

* No longer is there a multiple schema conflict on "colorScheme" (#15748) (thanks @brandondong!)

## Miscellaneous

* The WPF control now delay-loads UIAutomationCore.dll because it's incomplete in Windows RS1 (#15614)

With additional thanks to @floh96, @DaveRayment, @ushuz, and @LitoMore for documentation updates and @tusharsnx,
@Jvr2022, @Dan-Albrecht, @ebarnabas644, @jamespack for build system, project config, and compiler compatibility fixes.

[Windows Terminal Distributions]: https://learn.microsoft.com/windows/terminal/distributions

---

_Binary files inside the unpackaged distribution archive bear the version number `1.18.230925001`._

### Asset Hashes

- Microsoft.WindowsTerminal_1.18.2681.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip
   - SHA256 `BCAB5079B9F4E2ADBDA4B81D7482BA20BCBDA83B7BB77D7FFDBFF006234CE2A3`
- Microsoft.WindowsTerminal_1.18.2681.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `33A931384373DEEAF0E11F3E7BA985349FD2C32137DED81F9770F428558A76ED`
- Microsoft.WindowsTerminal_1.18.2681.0_arm64.zip
   - SHA256 `C4D3D18A184FD0B6A246F2307A0D3FA6B869F9D5670B3161E490F2E520F569E7`
- Microsoft.WindowsTerminal_1.18.2681.0_x64.zip
   - SHA256 `F4FE48C51B96B591B56B60A437ACDF73AA2797A01B77360660BDB47E1A9ED184`
- Microsoft.WindowsTerminal_1.18.2681.0_x86.zip
   - SHA256 `7BF1A048FED827D8638DE934F5DE8EB711CEF42E0CE46E0A638E0AC87AE56D64`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.18.2681.0'>Windows Terminal v1.18.2681.0</a>.</em>

**Comment by Almighty188** [comment]
@DHowett 
In the release notes above, it mentions:
"Windows Terminal 1.18 is here, and with it comes the coolest set of features ever:

You can finally pull tabs out of Terminal windows! You can also put other tabs back into other Terminal
windows. It's a whole thing."

I am using v1.18.2822.0 as Microsoft Store never downloaded  v1.18.2681 but it seems I am unable to pull the tab out of Windows Terminal windows as the tab will not drag even when I select it or am I doing it incorrectly?  Is rearranging the tabs in Windows Terminal also now possible?

**Reply by zadjii-msft** [reply]
(for future readers: we resolved this in #16126. OP was running the Terminal elevated, where drag-drop doesn't work. For more info, see #6661)

**Reply by Almighty188** [reply]
Yeah, I read that.  I didn't realize that it was a old issue as the release notes made it sound like a new feature in 1.18 but how exactly do  I do the following? 
"Now, on the bright side, there is a workaround for moving tabs without using drag/drop. There's a pair of actions in the Command Palette for "Move tab forward" / "Move tab backward" which will reorder your tabs without the need to invoke the drag/drop service. That might be a possible workaround."  as I have read the docs linked but still am unclear how to do it.

**Reply by zadjii-msft** [reply]
![c02bb97d-c2ad-407b-bd96-ff4779ccda2e](https://github.com/microsoft/terminal/assets/18356694/8b08bdf9-2d5c-41dc-bed6-658fb4607c0c)


(the command palette is bound to <kbd>ctrl+shift+p</kbd> by default)

**Reply by Almighty188** [reply]
Thanks, was trying to figure out what the keys in question was to get to the command palette.  Guess it will take some practice to be able to do this rapidly like you do on the keyboard.  Thanks again!
